api/dp-openapi.yaml

```diff
@@ -678,6 +678,474 @@ paths:
         default:
           $ref: '#/components/responses/UnexpectedError'
 
+  /projects/{projectid}/checkout:
+    put:
+      tags:
+      - project
+      description: Check out some pages for proofreading
+      parameters:
+      - name: projectid
+        in: path
+        description: ID of project
+        required: true
+        schema:
+          type: string
```

Comment:
should we create a custom project type since it often ends up in query params?

Reply:
Looks like we could use pattern to futher enhance the strength of the type:
https://swagger.io/docs/specification/data-models/data-types/#pattern
and there are a few other places in the swagger we could use the new ref projectid type if we want.

Reply:
Swagger is able to generate code although we are not using it for that purpose. Presumably the pattern would get built into the code. There is a function `validate_projectID()` in `pinc/Project.inc` which validates a projectid using a regular expression but the validator for projectid in `api/v1_validators.inc` does not use it but uses the function `_load_from_db()` which either loads the project or throws a nonexistentProject exception. Perhaps it would be useful to use `validate_projectID()` in our API validator. What does @cpeel  think?